tests/integration.rs

```diff
@@ -199,6 +200,34 @@ fn contents(#[case] harness: Harness) {
 	assert_eq!(entries, expected);
 }
 
+#[apply(all_images)]
+fn readdir_large(#[case] harness: Harness) {
+	let d = &harness.d;
+
+	let mut dir = nix::dir::Dir::open(
+		&d.path().join("large"),
+		OFlag::O_DIRECTORY | OFlag::O_RDONLY,
+		Mode::empty(),
+	)
+	.unwrap();
+
+	let mut entries = dir
+		.iter()
+		.map(|x| x.unwrap())
```

Comment:
Rust generates separate code for each closure, even for closures that are identical, so it's more efficient to write like this:
```suggestion
		.map(Result::unwrap)
```
Or, you can combine this `.map` with the following line.  Better yet, you can combine all three lines into a single `.filter_map` call.